src/index.ts

```diff
@@ -146,7 +150,7 @@ class RenderPDF {
         renderer.killChrome();
     }
 
-    async renderPdf(url: string, options: RenderOptions) {
+    async renderPdf(url: string, options: RenderOptions & ThriveRenderPDFOptions) {
```

Comment:
question - instead of a boolean - would it be possible to make it so that the event name is actually passed through, and the caller can specify the event they would like to wait for? e.g.

`waitForEvent=thrive.jsdone`

instead of 

`delayUntilThriveEventFires=true`

Reply:
Perhaps more versatile would be a --no-automatic-print flag, and then just wait for the js to call `print()`

Reply:
I have looked into the headless render code and it's doing a lot for us that `print()` doesn't cover.
I'll retract my solution in favour of `waitForEvent=thrive.jsdone` which I think is really promising.

Reply:
> question - instead of a boolean - would it be possible to make it so that the event name is actually passed through, and the caller can specify the event they would like to wait for? e.g.
> 
> `waitForEvent=thrive.jsdone`
> 
> instead of
> 
> `delayUntilThriveEventFires=true`

I suppose so, would make it more OSS friendly if we ever wanted to promote this change upstream (i don't think we do). Making it very thrive specific makes it clear what callers need to do and avoid someone accidentally firing events that clash in name with another event. 

> Perhaps more versatile would be a --no-automatic-print flag, and then just wait for the js to call print()

Not sure how this would work. To clarify, the current flow is:
html is loaded -> renderPDF setsup the listener -> js in the html runs -> after js execution finishes and fires the event -> listener picks up the event and resolves the promise, continuing the pdf'ing

You'd need your proposed print method back into the html and while there are some mechanism for doing this (i.e. setting to window perhaps, you'd now need to make sure your JS library knows about window as a global namespace) i'm not sure that's as nice as using existing message passing systems in js already (i.e. events)